contracts/contracts/EthereumUtils.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.0;
+
+library EthereumUtils {
+    uint256 internal constant K256_P =
+        0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2f;
+
+    // (p+1)//4
+    uint256 internal constant K256_P_PLUS_1_OVER_4 =
+        0x3fffffffffffffffffffffffffffffffffffffffffffffffffffffffbfffff0c;
+
+    address internal constant PRECOMPILE_BIGMODEXP = address(0x5);
+
+    error expmod_Error();
```

Comment:
Have you ever encountered a `revert Error()` on Sapphire? it's gas efficient but the developer experience is bad. try it out and see why I have these [weird comments](https://github.com/escrin/escrin/blob/main/evm/contracts/enclave-identity/Lockbox.sol#L9) in my contracts :P

Reply:
With Ethers v6 it seems fairly straightforward.

I can double check if there's any difference with Sapphire, but I'm fairly sure I've used them with Sapphire before.

```
   } catch (e) {
    if (e.data && contract) {
      const decodedError = contract.interface.parseError(e.data);
      console.log(`Transaction failed: ${decodedError?.name}`);
    } else {
      console.log(`Error:`, e);
    }
  }
```